File: kotlin/0080-remove-duplicates-from-sorted-array-ii.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {

        var left = 0
        var right = 0

        while (right < nums.size) {
            var count = 1
            while(right + 1 < nums.size && nums[right] == nums[right + 1]) {
                count++
                right++
            }
            for(i in 0 until minOf(2, count)) {
                nums[left] = nums[right]
                left++
            }
            right++
        }

        return left
    }
}

File: csharp/0080-remove-duplicates-from-sorted-array-ii.cs
public class Solution
{
    public int RemoveDuplicates(int[] nums)
    {
        int l = 0;
        int r = 0;

        while ( r < nums.Length)
        {
            int count = 1;

            while (r + 1 < nums.Length && nums[r] == nums[r + 1])
            {
                count++;
                r++;
            }

            for (int i = 0; i < Math.Min(2, count); i++)
            {
                nums[l] = nums[r];
                l++;
            }

            r++;
        }

        return l;
    }
}